one_fm/operations/doctype/employee_checkin_issue/employee_checkin_issue.py

```diff
@@ -60,70 +41,62 @@ def validate_duplicate_record(self):
 		date = getdate(self.date).strftime('%d-%m-%Y')
 		if frappe.db.exists("Employee Checkin Issue", {"employee": self.employee, "date":self.date,
 			"assigned_shift": self.assigned_shift, "log_type": self.log_type, "name": ["not in", [self.name]]}):
-			frappe.throw(_("{employee} has already created a Employee Checkin Issue for {log_type} on {date}.".format(employee=self.employee_name, log_type=self.log_type, date=date)))
-
-	@frappe.whitelist()
-	def create_issue(self):
-		if not self.issue:
-			issue_type = False
-			if frappe.db.exists('Issue Type', {'name': 'Checkin Issue'}):
-				issue_type = 'Checkin Issue'
-			else:
-				issue_type_doc = frappe.get_doc({"doctype": "Issue Type", "__newname": "Checkin Issue"}).insert()
-				issue_type = issue_type_doc.name
-			if issue_type:
-				issue = frappe.new_doc('Issue')
-				issue.subject = "Employee Checkin Issue - {0}".format(self.issue_type)
-				issue.raised_by = frappe.session.user
-				doc_link = frappe.utils.get_url(self.get_url())
-				description = issue.subject + "<br/><p>The user found an issue in the app \
-				 	and recorded in <a href='{0}'>Employee Checkin Issue</a>.</p>".format(doc_link)
-				issue.description = description
-				issue.issue_type = issue_type
-				issue.save(ignore_permissions=True)
-				self.issue = issue.name
-				self.save(ignore_permissions=True)
-				self.reload()
+			msg = _(
+				"{employee} has already created a Employee Checkin Issue for {log_type} on {date}.".format(
+					employee=self.employee_name, log_type=self.log_type, date=date
+				)
+			)
+			frappe.throw(msg)
 
-	def on_update(self):
-		if self.workflow_state == 'Approved':
-			create_employee_checkin_for_employee_checkin_issue(self)
-			workflow_approve_reject(self, [get_employee_user_id(self.employee)])
-
-		if self.workflow_state == 'Pending':
-			send_workflow_action_email(self, recipients=[get_employee_user_id(self.shift_supervisor)])
+	def validate_attendance(self):
+		attendance = frappe.db.exists('Attendance',{'attendance_date': self.date, 'employee': self.employee, 'docstatus': 1})
+		if attendance:
+			frappe.throw(_('There is an Attendance {0} exists for the \
+			Employee {1} on {2}'.format(attendance, self.employee_name, format_date(self.date))), exc=ExistAttendance)
 
-		if self.workflow_state in ['Rejected']:
-			workflow_approve_reject(self, [get_employee_user_id(self.employee)])
+	def validate_employee_checkin(self):
+		start_date = get_datetime(self.date)
+		end_date = add_to_date(start_date, hours=23.9998)
+		employee_checkin = frappe.db.exists(
+			"Employee Checkin",
+			{
+				"log_type": self.log_type,
+				"time": ["between", [start_date, end_date]],
```

Comment:
And Shift Assignment needs to be considered here. Since on can have multiple shifts (Basic And OT).